### PR TITLE
feat: Add visualization on top new standardization ontology layer

### PR DESCRIPTION
**Pull request by wsy19961129**

### CODE REVIEW

File: packages/itmat-commons/src/models/log.ts [hunk 1]
```diff
@@ -68,16 +68,13 @@ export enum LOG_ACTION {
     createFieldCurationJob = 'CREATE_FIELD_CREATION_JOB',
     createDataCurationJob = 'CREATE_DATA_CURATION_JOB',
     getDataRecords = 'GET_DATA_RECORDS',
-    getOntologyTree = 'GET_ONTOLOGY_TREE',
```
Comment (fguitton) [inline packages/itmat-commons/src/models/log.ts:71]: You cannot remove these here because past records in the database might still exist and be valid ⚠️

**Review by fguitton — COMMENTED** [review]
Generally there is a lot of hardcoding going on.
In the background I can see a progressive removal of DMP agnostic behavior to the benefit of a very CDISC focused approach. This was not the original philosophy.
@wsy19961129 Can you provide context on how this is required and how we mitigate in the future ?

**Comment by wsy19961129** [comment]
> Generally there is a lot of hardcoding going on. In the background I can see a progressive removal of DMP agnostic behavior to the benefit of a very CDISC focused approach. This was not the original philosophy. @wsy19961129 Can you provide context on how this is required and how we mitigate in the future ?

Because when we want to visualize the fields (e.g., age, race) we have to

File: packages/itmat-ui-react/src/components/projectDetail/tabContent/utils/defaultParameters.ts [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+            },
+            {
+                visit: '2',
+                race: [],
+                genderID: [],
+                siteID: [],
+                age: [0, 100],
+                mh: [],
+                filters: [{
+                    field: '3075', op: '>=', value: '60'
```
[the remark this answers is not in the document]
Comment (wsy19961129) [reply packages/itmat-ui-react/src/components/projectDetail/tabContent/utils/defaultParameters.ts:105]: > Generally there is a lot of hardcoding going on. In the background I can see a progressive removal of DMP agnostic behavior to the benefit of a very CDISC focused approach. This was not the original philosophy. @wsy19961129 Can you provide context on how this is required and how we mitigate in the future ?

For the frontend, hard-coding the fieldId is used for:
1. While visualizing the demographics, we don't ask for any user inputs, and we only want to visualize some of the fields within the demographics, thus we have to know the original fieldIds. 
2. The analysis tab provides several templates whose fieldIds are already filled in.

Based on current APIs (get data through fieldIds), we have to specify the fieldIds, do you have any better ideas on improving this?

For the backend, 
As we provide several output formats:
    1. _raw_ for original data;
    2. _summary_ and _grouped_ for statistics;
    3. _standardized_ for CDISC format.

I think  **3** is indeed designed specifically for CDISC output. but **1 & 2** are in common use for any clinical dataset on DMP. 

If we are not enforcing the CDISC format, then at which stage should we do the CDISC standardization?

File: packages/itmat-commons/src/graphql/projects.ts [hunk 3]
```diff
@@ -11,6 +11,14 @@ export const GET_PROJECT = gql`
...
             approvedFields @include(if: $admin)
             approvedFiles @include(if: $admin)
+            dataVersion {
+                id
+                version
+                contentId
+                tag
+                updateDate
+            }
+            summary
```
[the remark this answers is not in the document]
Comment (wsy19961129) [reply packages/itmat-commons/src/graphql/projects.ts:21]: Yes, the _dataVersion_ is obtained from the associated study and not a fixed value. So it won't effect the old projects.

https://github.com/ideafast/ideafast-portal/blob/747687ce9aee17c26d15577745c036827d9cb3ea/packages/itmat-interface/src/graphql/resolvers/studyResolvers.ts#L355-L364

**Comment by wsy19961129** [comment]
@fguitton Could you have a review on the current commit? If there are no main issues, I will merge the conflicts and update it then,

**Pull request merged**